### DISCUSSION TITLE
Building new Moose Action

### DISCUSSION THREAD

**Discussion by AmrIbrahim8894** [idaholab/moose#27411] [Q&A General]
Hi there,
I am trying to build a custom cohesive zone action called `PoroCohesiveZoneAction` for variables (fluid velocity, pressure) besides displacements and tractions in our `farmsApp`. Following the code in `SolidMechanics`, We register our action in `farmsApp.C`:

```
static void
associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
{
registerSyntax("PoroCohesiveZoneAction", "Actions/*");
}
```
And in the input file, we call the action:
```
[Actions/PoroCohesiveZoneAction]
    [./czm_ik]
        boundary = 'Block0_Block1'
        strain = SMALL
        generate_output='traction_x traction_y jump_x jump_y jump_vel_x jump_vel_y pressure_plus pressure_minus normal_traction tangent_traction normal_jump tangent_jump'
    [../]
[]
```
However, MOOSE is unable to find the action, returns the error:
```
*** ERROR ***
/home/amr/projects/farms/examples/benchmark_tpv2052D/version1/tpv2052D_Poro.i:84.1: section '[Actions]' does not have an associated "Action".
 Common causes:
- you misspelled the Action/section name
- the app you are running does not support this Action/syntax
/home/amr/projects/farms/examples/benchmark_tpv2052D/version1/tpv2052D_Poro.i:84.1: section '[Actions/PoroCohesiveZoneAction]' does not have an associated "Action".
 Common causes:
- you misspelled the Action/section name
- the app you are running does not support this Action/syntax
/home/amr/projects/farms/examples/benchmark_tpv2052D/version1/tpv2052D_Poro.i:85.5: section '[Actions/PoroCohesiveZoneAction/czm_ik]' does not have an associated "Action".
 Common causes:
- you misspelled the Action/section name
- the app you are running does not support this Action/syntax
```
I appreciate your help. The MOOSE I used is not the most updated version, although I don't think this is the reason of the problem. 
And a follow-up question is: how to link this custom cohesive zone to the original cohesive zone model, since we would like to solve displacements and tractions also.

**Comment by GiudGiud** [comment]
Hello

The syntax used seems to be this instead
```
registerSyntax("PoroCohesiveZoneAction", "Actions/PoroCohesiveZoneAction/*");
```

**Comment by AmrIbrahim8894** [comment]
I did that but nothing change, it still show the same error.

**Reply by GiudGiud** [reply]
did you recompile after changing? I m asking because you replied very soon after my post that's it

**Comment by AmrIbrahim8894** [comment]
yes. i recompiled then i tried to run the problem again. should i do also make clobberall ?

**Reply by GiudGiud** [reply]
should not be necessary. 
ok using your executable, use the --dump or --json followed by &> some_file_name.txt to examine all the registered syntax

try to find what PoroCohesiveZoneAction got registered to there

**Reply by AmrIbrahim8894** [reply]
Sorry, I don't know how to use that --dump or --json followed by &> some_file_name.txt. Can you provide example command ? Thank you

**Reply by GiudGiud** [reply]
```
your_executable-opt --dump &> dump_file.txt
```

**Reply by AmrIbrahim8894** [reply]
It gave this error:
```
[31m
*** ERROR ***
basic_string::_M_construct null not valid[39m

Stack frames: 7

0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<char const*>(char const*&&)
3: /home/amr/projects/moose/framework/libmoose-opt.so.0(+0x9686cf) [0x7fa8e20b86cf]
4: main
5: __libc_start_main
6: ./farms-opt(+0x33ef) [0x56390badf3ef]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
```
what should i do ?

**Reply by GiudGiud** [reply]
I dont know about that one.
what about --json

**Reply by AmrIbrahim8894** [reply]
I tried --json but it gave me the same error.

**Reply by GiudGiud** [reply]
ugh this is unexpected. 
what did you type exactly?
and if you build in moose/test can you perform a syntax dump with the moose_test executable?
